train.lua

```diff
@@ -32,349 +35,195 @@ cmd:text('Train a character-level language model')
 cmd:text()
 cmd:text('Options')
 -- data
-cmd:option('-data','tinyshakespeare','name of data directory. Should contain the file input.txt with input data')
-cmd:option('-back','cuda','cpu|cuda|cl')
-cmd:option('-len', -1, 'only use this many characters from input data. -1 for all data')
+cmd:option('-data', 'tinyshakespeare', 'name of data directory. Should contain the file input.txt with input data')
+cmd:option('-back', 'cuda', 'cpu|cuda|cl')
 cmd:option('-seq', 50, 'sequence length to use')
-cmd:option('-dump', 100, 'iterations between writing out weights')
```

Comment:
This option is at least useful for unit testing / sanity testing.  There should be a way of training for just 3-10 seconds, and then sampling from that, and checking that it all kind of worked ok, at least, didnt crash.